src/components/layout/layout.js

```diff
@@ -16,22 +17,77 @@ const Main = styled.main`
     position: relative;
 `
 
+const is_browser = typeof window !== 'undefined'
+const has_dataLayer = is_browser && window.dataLayer
+const cookie_expires = 7
+
 const Layout = ({ children, type, interim_type, padding_top, no_login_signup }) => {
+    const [clients_country, setClientCountry] = React.useState(Cookies.get('clients_country'))
+    const [show_cookie_banner, setShowCookieBanner] = React.useState(false)
+    const [has_window_loaded, setWindowLoaded] = React.useState(false)
+
     const is_static = type === 'static'
 
-    // Handle cookie banners
-    const [show_cookie_banner, setShowCookieBanner] = React.useState(false)
     React.useEffect(() => {
-        const clients_country = Cookies.get('clients_country')
-        if (isEuCountry(clients_country)) {
-            const has_cookie = Cookies.get('has_cookie_accepted')
-            setShowCookieBanner(!has_cookie)
+        if (!clients_country) {
+            const binary_socket = BinarySocketBase.init()
+
+            binary_socket.onopen = () => {
+                binary_socket.send(JSON.stringify({ website_status: 1 }))
+            }
+
+            binary_socket.onmessage = (msg) => {
+                const response = JSON.parse(msg.data)
+
+                if (!response.error) {
+                    setClientCountry(response.website_status.clients_country)
+                    Cookies.set('clients_country', response.website_status.clients_country, {
+                        expires: cookie_expires,
+                    })
+                }
+
+                binary_socket.close()
+            }
+        }
+
+        const windowLoaded = () => setWindowLoaded(true)
+
+        if (is_browser) {
+            window.addEventListener('load', windowLoaded)
         }
+
+        return () => window.removeEventListener('load', windowLoaded)
     }, [])
+
+    React.useEffect(() => {
+        if (!clients_country || !has_window_loaded) return
+
+        const is_eu_country = isEuCountry(clients_country)
```

Comment:
means this will be always called on every page load then?

Reply:
it will be called once when window has load and the client has a country  - (changing route in gatsby is not counted as a window load event)

Reply:
@mustofa-binary moved has_window_loaded to our global context instead